atd-toolbox/data_model/python/sql_templates/crashes_list_view.sql

```diff
@@ -241,3 +241,70 @@ left join
 left join
     lookups.injry_sev_lkp
     on lookups.injry_sev_lkp.id = crash_injury_metrics_view.crash_injry_sev_id;
+
+
+create materialized view locations_list_view as (
+    with crash_totals as (
+        with unioned_crash_counts as (
+            with cr3_crash_counts as (
+                select
+                    location_id,
+                    sus_serious_injry_count,
+                    vz_fatality_count,
+                    est_comp_cost_crash_based
+                from
+                    crashes_list_view
+                where
+                    crashes_list_view.private_dr_fl = false
+                    and crashes_list_view.location_id is not null
+                    and crashes_list_view.crash_date
+                    > (now() - '5 years'::interval)
+            ),
+
+            non_cr3_crash_counts as (
+                select
+                    location_id,
+                    0 as vz_fatality_count,
+                    0 as sus_serious_injry_count,
+                    est_comp_cost as est_comp_cost_crash_based
+                from atd_apd_blueform as non_cr3_crash_counts
+                where
+                    true
+                    and non_cr3_crash_counts.location_id is not null
+                    and non_cr3_crash_counts.date
+                    > (now() - '5 years'::interval)
+            )
+
+            select * from cr3_crash_counts
+            union all
+            select * from non_cr3_crash_counts
+        )
+
+        select
+            location_id,
+            count(unioned_crash_counts.*) as crash_count,
+            sum(
+                unioned_crash_counts.sus_serious_injry_count
+            ) as sus_serious_injry_count,
+            sum(unioned_crash_counts.vz_fatality_count) as vz_fatality_count,
+            sum(
+                unioned_crash_counts.est_comp_cost_crash_based
+            ) as total_est_comp_cost
+        from
+            unioned_crash_counts
+        group by location_id
+    )
+
+    select
+        locations.location_id,
+        locations.description,
+        coalesce(crash_totals.crash_count, 0) as crash_count,
+        coalesce(
+            crash_totals.sus_serious_injry_count, 0
+        ) as sus_serious_injry_count,
```

Comment:
do these counts need to be coalesced? i don't think there is harm in it and i'm mostly thinking through if any of the unioned records can ever not be non-integer

Reply:
take my answer with a grain of salt, but we're making sure that no `null` values work their way into these rollups, where just one `null` value has the affect of making the entire count `null`. this can happen when a crash for whatever reason does not have unit records, or a unit for whatever reason does not have people records

Reply:
gotcha - that makes sense to me and i see how that could come into play now - thanks!